src/tests/run.sh

```diff
@@ -11,7 +11,7 @@ function print_usage {
     echo 'Optional arguments:'
     echo '  -h|--help                        : Show usage information.'
     echo '  -v, --verbose                    : Show output from each test.'
-    echo '  <arch>                           : One of x64, x86, arm, arm64, loongarch64, riscv64, wasm. Defaults to current architecture.'
+    echo '  <arch>                           : One of x64, x86, arm, arm64, loongarch64, riscv64, wasm. Defaults to x64.'
```

Comment:
This is not correct, it does default to the current architecture. Only if we don't recognize the architecture does it default to x64.

Reply:
it is ambiguous.

Reply:
Not really, we check `uname -m`

Reply:
@akoeplinger, in `eng/native/init-os-and-arch.sh`, we set `arch` to `x64` in default case (when all the known cases fail). So this modified comment is correct.

Reply:
@am11 yes but only if known cases fail. If I run the build on an ARM64 machine then it will default to ARM64, not x64, so the comment is incorrect.

Reply:
Yup, from usage of run.so it assigns current arch unless uname -m gives unknown value which we haven’t handled. So let’s revert this line.

Reply:
@shushanhf, you can change this: https://github.com/dotnet/runtime/blob/04690205c07a07a242155e88fff0acb730f3dc73/eng/native/init-os-and-arch.sh#L77-L78 to:

```sh
echo "Unknown CPU $CPUName detected,"
exit 1
```

because we don't want to proceed the build for people porting the runtime to new architecture.

Reply:
Sorry for reponse late.
yes, I also want to set the default as error exit when unknow CPU.
There is no need to continue for unknow architecuture.

Reply:
Hi, @am11  @akoeplinger 
I had updated the patch, is it OK ?
Thanks